### ISSUE TITLE
Can't login via Google anymore

### ISSUE THREAD

**Issue by devbyray** [supabase/supabase#13192]
# Bug report

1. Go to https://generatorxyz.com
2. Login with Google account which isn't already registered
3. Redirect looks like: https://www.generatorxyz.com/?error_description=Database+error+saving+new+user&error=server_error
4. The error from the log looks like this:

```
failed to close prepared statement: ERROR: current transaction is aborted, commands ignored until end of transaction block (SQLSTATE 25P02): ERROR: column \"user_id\" is of type uuid but expression is of type text (SQLSTATE 42804)",
```

The only place where I use a trigger function with the "user_id" is by generating credits for new users. That function looks like this:

```
create function public.handle_new_user_credits() 
returns trigger as $$
begin
  insert into public.credits (id, user_id, amount)
  values (new.id, new.raw_user_meta_data->>'id', 30);
  return new;
end;
$$ language plpgsql security definer;
```

This worked before, but now it doesn't 🤔

Can someone help me with this?

**Comment by toff26** [comment]
Hi @devbyray, I was able to sign in to the website seamlessly, and the URL had access_token in it.

**Comment by devbyray** [comment]
True, that is because I had to turn off the function, otherwise no new users couldn’t login.

**Comment by GaryAustin1** [comment]
This does not appear to be a bug.  Moving to discussions.
Your error says you have a uuid user_id column and you are reading in json text.  Add ::UUID like:
`(new.raw_user_meta_data->>'id')::UUID` is most likely your fix.
